### PR TITLE
ESLint config changes and fixes

### PR DESCRIPTION
**Pull request by darrellwarde**
# Description

> **Note**
> 
> Please provide a description of the work completed in this PR below

This PR massively simplifies the ESLint config for the monorepo, and fixes a number of rule violations where appropriate.

### CODE REVIEW

**Comment by darrellwarde** [comment]
I have firmly broken the Toolbox build. 🙃

File: packages/graphql-toolbox/webpack.config.js [hunk 1]
```diff
@@ -17,24 +17,32 @@ module.exports = {
     context: path.join(__dirname),
     target: "web",
     resolve: {
-        plugins: [new TsconfigPathsPlugin()],
+        plugins: [new TsconfigPathsPlugin({ configFile: "./tsconfig.production.json" })],
```
Comment (tbwiss) [inline packages/graphql-toolbox/webpack.config.js:20]:
```suggestion
        plugins: [new TsconfigPathsPlugin({ configFile: "tsconfig.production.json" })],
```

File: packages/graphql-toolbox/tsconfig.json [hunk 2]
```diff
@@ -1,21 +1,14 @@
...
-        "noImplicitAny": false,
-        "rootDir": "src",
+        "rootDir": ".",
         "baseUrl": ".",
         "outDir": "dist",
-        "noEmit": false,
         "paths": {
             "@neo4j/graphql": ["../graphql/src"],
-            "@neo4j/introspector": ["../introspector/src"]
+            "@neo4j/graphql-plugin-auth": ["../plugin/auth/graphql-plugin-auth/src"]
```
Comment (tbwiss) [inline packages/graphql-toolbox/tsconfig.json:9]: I think the auth plugin is not needed in this file

**Review by tbwiss — COMMENTED** [review]
Two things I saw which might fix the toolbox build!

**Comment by neo4j-team-graphql** [comment]
Made some changes to the GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at
 http://neo4j-graphql-toolbox-stage-2131.surge.sh (=> for local development; use for instance [Neo4j Desktop](https://neo4j.com/download/))
or
 https://neo4j-graphql-toolbox-stage-2131.surge.sh (=> use to connect to an [Aura Free](https://neo4j.com/cloud/aura-free/) dbms)

File: package.json [hunk 3]
```diff
@@ -27,14 +27,15 @@
...
     },
     "devDependencies": {
-        "@tsconfig/node14": "1.0.3",
+        "@tsconfig/node16": "^1.0.3",
         "@typescript-eslint/eslint-plugin": "5.39.0",
         "@typescript-eslint/parser": "5.39.0",
         "concurrently": "7.4.0",
         "dotenv": "16.0.3",
         "eslint": "8.24.0",
         "eslint-config-airbnb-typescript": "17.0.0",
```
Comment (angrykoala) [inline package.json:36]: Are we still using airbnb eslint?

**Review by angrykoala — APPROVED** [review]
Looks good :clap:

**Pull request merged**